nucypher/config/configs.py

```diff
@@ -64,34 +43,113 @@ def db_path(self):
         return self.__db_path
 
 
+class BlockchainConfig:
+
+    __default_providers = (IPCProvider(ipc_path=os.path.join('/tmp/geth.ipc')),
+                           # user-managed geth over IPC assumed
+                           )
+
+    __default_network = 'tester'
+    __default_timeout = 120          # seconds
+    __configured_providers = list()  # tracks active providers
+
+    def __init__(self, network: str=None, timeout: int=None,
+                 compiler=None, registrar=None, deploy=False,
+                 geth=True, tester=False):
+
+        # Parse configuration
+
+        if len(self.__configured_providers) == 0:
+            warnings.warn("No blockchain provider backends are configured, using default.", RuntimeWarning)
+            self.__providers = BlockchainConfig.__default_providers
+
+        self._providers = self.__configured_providers
+        self.__network = network if network is not None else self.__default_network
+        self.__timeout = timeout if timeout is not None else self.__default_timeout
+
+        if deploy is False:
+            from nucypher.blockchain.eth.interfaces import ContractInterface
+            interface_class = ContractInterface
+        else:
+            from nucypher.blockchain.eth.interfaces import DeployerInterface
+            interface_class = DeployerInterface
+
+        interface = interface_class(blockchain_config=self, sol_compiler=compiler, registrar=registrar)
+
+        if geth is True:
+            w3 = interface.w3
+            w3.middleware_stack.inject(geth_poa_middleware, layer=0)
```

Comment:
Well, it's POA until it's not. I can certainly see this class being used with real blockchain (even on testnet!) where it's not POA anymore

Reply:
Agreed, this either needs to move or be renamed.